### PR TITLE
Fix #5796: Format sol and spl token amount with their token decimal value.

### LINKED ISSUE #5796: Wallet Bug: Formatting SPL Token amount is not respecting token decimals in sending transaction
### Description: 
Wrong formatting send SPL token amount in sending

### Steps to Reproduce 
  1. send some amount of SPL token (make sure this SPL token decimal is not 9)
  2. check the confirmation screen

**Actual result:** 
the send amount is wrong
1. it will be much bigger if the spl token decimals is smaller than 9
2. it will be much smaller if the spl token decimals is bigger than 9

**Expected result:**
the send amount should be the same as user input in send screen

**Reproduces how often:** [Easily reproduced]

### PR DESCRIPTION
**Pull request by nuo-xu**
This pull request fixes #5796

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [ ] Light & dark mode
  - [ ] Different size classes (iPhone, landscape, iPad)
  - [ ] Different dynamic type sizes

## Test Plan:
please refer to the issue

Simulator Screen Shot - iPhone Xʀ - 2022-08-05 at 13 52 05 | Simulator Screen Shot - iPhone Xʀ - 2022-08-05 at 13 53 15
--|--
Simulator Screen Shot - iPhone Xʀ - 2022-08-05 at 13 54 15 | Simulator Screen Shot - iPhone Xʀ - 2022-08-05 at 13 54 32
--|--

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: BraveWallet/Extensions/WeiFormatter.swift [hunk 1]
```diff
@@ -150,13 +150,13 @@ struct WeiFormatter {
       .rounded().asString(radix: outputRadix.rawValue)
   }
   
-  static func decimalToLamports(_ decimalString: String) -> UInt64? {
+  static func decimalToSplAmount(_ decimalString: String, tokenDecimals: Int) -> UInt64? {
```
Comment (StephenHeaps) [inline BraveWallet/Extensions/WeiFormatter.swift]: I don't think we should include `Spl` in this function name as it's used for SOL tokens too

File: Tests/BraveWalletTests/SendTokenStoreTests.swift [hunk 2]
```diff
@@ -356,4 +356,108 @@ class SendTokenStoreTests: XCTestCase {
...
+      defer { ex.fulfill() }
+      XCTAssertTrue(success)
+    }
+    
+    waitForExpectations(timeout: 3) { error in
+      XCTAssertNil(error)
+    }
+  }
+  
+  func testSendSolTokenAmount() {
```
Comment (StephenHeaps) [inline Tests/BraveWalletTests/SendTokenStoreTests.swift]: `testSendSolTokenAmount()` -> `testSendSplTokenAmount()`

File: Tests/BraveWalletTests/SendTokenStoreTests.swift [hunk 3]
```diff
@@ -356,4 +356,108 @@ class SendTokenStoreTests: XCTestCase {
...
+    let rpcService = BraveWallet.TestJsonRpcService()
+    rpcService._chainId = { $1(BraveWallet.NetworkInfo.mockSolana.chainId) }
+    rpcService._network = { $1(BraveWallet.NetworkInfo.mockSolana)}
+    rpcService._solanaBalance = { _, _ , completion in
+      completion(UInt64(mockBalance), .success, "")
+    }
+    rpcService._addObserver = { _ in }
+    
+    let walletService = BraveWallet.TestBraveWalletService()
+    walletService._userAssets = { $2([.previewToken]) }
```
Comment (StephenHeaps) [inline Tests/BraveWalletTests/SendTokenStoreTests.swift]: We should use `.mockSolToken, .mockSpdToken` here for solana tests (`.previewToken` is ETH) 🙂

**Review by iccub — APPROVED** [review]

**Review by StephenHeaps — COMMENTED** [review]
Thanks for adding unit tests for bug 👍

File: BraveWallet/Extensions/WeiFormatter.swift (hunk 1 above)
Comment (StephenHeaps) [reply BraveWallet/Extensions/WeiFormatter.swift]: Resolved in [f30db1f](https://github.com/brave/brave-ios/pull/5797/commits/f30db1f20bc465c71ab738a1de129920e79622b6)

File: Tests/BraveWalletTests/SendTokenStoreTests.swift (hunk 3 above)
Comment (StephenHeaps) [reply Tests/BraveWalletTests/SendTokenStoreTests.swift]: Resolved in [f30db1f](https://github.com/brave/brave-ios/pull/5797/commits/f30db1f20bc465c71ab738a1de129920e79622b6)

File: Tests/BraveWalletTests/SendTokenStoreTests.swift (hunk 2 above)
Comment (StephenHeaps) [reply Tests/BraveWalletTests/SendTokenStoreTests.swift]: Resolved in [f30db1f](https://github.com/brave/brave-ios/pull/5797/commits/f30db1f20bc465c71ab738a1de129920e79622b6)

**Review by StephenHeaps — APPROVED** [review]

**Pull request merged**